tempesta_fw/t/unit/sched_helper.c

```diff
@@ -133,3 +134,82 @@ test_conn_release_all(TfwSrvGroup *sg)
 		}
 	}
 }
+
+void test_sched_generic_empty_sg(struct TestSchedHelper *sched_helper)
+{
+	size_t i;
+	TfwSrvGroup *sg;
+
+	BUG_ON(!sched_helper);
+	BUG_ON(!sched_helper->sched);
+	BUG_ON(!sched_helper->conn_types);
+	BUG_ON(!sched_helper->get_sched_arg);
+	BUG_ON(!sched_helper->free_sched_arg);
+
+	sg = test_create_sg("test", sched_helper->sched);
+
+	for (i = 0; i < sched_helper->conn_types; ++i) {
+		TfwMsg *msg = sched_helper->get_sched_arg(i);
+		TfwConnection *conn = sg->sched->sched_srv(msg, sg);
+
+		EXPECT_NULL(conn);
+		sched_helper->free_sched_arg(msg);
+	}
+
+	test_sg_release_all();
+}
+
+void test_sched_generic_one_srv_in_sg_zero_conn(struct TestSchedHelper *sched_helper)
+{
+	size_t i;
+	TfwSrvGroup *sg;
+
+	BUG_ON(!sched_helper);
+	BUG_ON(!sched_helper->sched);
+	BUG_ON(!sched_helper->conn_types);
+	BUG_ON(!sched_helper->get_sched_arg);
+	BUG_ON(!sched_helper->free_sched_arg);
+
+	sg = test_create_sg("test", sched_helper->sched);
+
+	test_create_srv("127.0.0.1", sg);
+
+	for (i = 0; i < sched_helper->conn_types; ++i) {
+		TfwMsg *msg = sched_helper->get_sched_arg(i);
+		TfwConnection *conn = sg->sched->sched_srv(msg, sg);
+
+		EXPECT_NULL(conn);
+		sched_helper->free_sched_arg(msg);
+	}
+
+	test_sg_release_all();
+}
+
+void test_sched_generic_max_srv_in_sg_and_zero_conn(struct TestSchedHelper *sched_helper)
```

Comment:
Too long line, please don't exceed 80 characters per line. In fact the function name is just enormous large. Also we write function type and its name on different lines. Please read https://github.com/tempesta-tech/linux-4.1.27-tfw/blob/master/Documentation/CodingStyle